projects/components/src/action-menu/action-search.provider.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Injectable } from '@angular/core';
+import { TranslationService } from '@vcd/i18n';
+import { ActionItem } from '../common/interfaces';
+import { SpotlightSearchProvider, SpotlightSearchResult, SpotlightSearchService } from '../spotlight-search';
+
+const ACTION_PROVIDER_SECTION_TITLE_KEY = 'vcd.cc.action.provider.section.title';
+
+@Injectable()
+export class ActionSearchProvider<R, T> implements SpotlightSearchProvider {
+    /**
+     * Initialized by the {@link ActionMenuComponent.actions} and is used for displaying actions on the {@link SpotlightSearchComponent}
+     */
+    actions: ActionItem<R, T>[];
+    /**
+     * Used in {@link isActionDisabled} to calculate disabled state of actions
+     */
+    selectedEntities: R[];
+
+    private _actionProviderName: string;
+    /**
+     * Set by the {@link ActionMenuComponent.actionProviderName} to display in the title of the action search results section as
+     * Actions: <actionProviderName>
+     */
+    set actionProviderName(val: string) {
+        if (!val || (val && val === this._actionProviderName)) {
+            return;
+        }
+        this._actionProviderName = val;
+        this.register();
+    }
+
+    private spotlightSearchProviderRegistrationId: string;
+
+    constructor(private spotlightSearchService: SpotlightSearchService, private ts: TranslationService) {}
+
+    /**
+     * Searches through nested actions and finds all the actions that match with entered search text on the
+     * {@link SpotlightSearchComponent}
+     */
+    search(criteria: string): SpotlightSearchResult[] {
+        if (!criteria) {
+            return [];
+        }
+
+        return this.findSearchCriteriaInNestedActions(criteria.toLowerCase()).map(action => ({
+            displayText: action.isTranslatable === false ? action.textKey : this.ts.translate(action.textKey),
+            handler: action.handler,
+        }));
+    }
+
+    /**
+     * Registers the actions with {@link SpotlightSearchService}. Called by {@link ActionMenuComponent}
+     */
+    register(): void {
+        if (this.spotlightSearchProviderRegistrationId) {
```

Comment:
As it is of now it seems that we cannot instantiate 2 ActionSearchProvider-s. If for some reason there are 2 components in the UI the second one that injects the ActionSearchProvider will unregister the previous one

Reply:
This provider is provided for each action menu component instance. Please see https://github.com/vmware/vmware-cloud-director-ui-components/pull/143/files#diff-1ae5d00fd1a984540f5a721b704076b7R32